src/tiling/plane_region.rs

```diff
@@ -115,6 +144,7 @@ pub struct PlaneRegion<'a, T: Pixel> {
   pub plane_cfg: &'a PlaneConfig,
   // private to guarantee borrowing rules
   rect: Rect,
+  max_rect: Rect,
```

Comment:
I already have concerns about the amount of data being passed around with PlaneRegions. You should create a separate struct for when you actually need this data.

Reply:
I'm not at all sure what the concern is... could you elaborate?  the 16 bytes for a rect saves a few large plane copies and provides an addiitonal measure of limits checking as now PlaneRegions can be used where before it's needed to copy into a frame or use slices (that can only enforce beginning/end).

The only decent way I see to keep this simple is to extend the abstraction transparently.  The goal here is to make mistakes harder.

Reply:
Oh!  Or do you mean 'PlaneRegion wasn't meant for this' and you're concerned it's being--- perverted....